interpreter/consul/src/main/scala/io/buoyant/interpreter/consul/ConsulDtabCache.scala

```diff
@@ -0,0 +1,91 @@
+package io.buoyant.interpreter.consul
+
+import com.google.common.cache.{CacheBuilder, CacheLoader}
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.finagle.{Dtab, Failure, Path}
+import com.twitter.io.Buf
+import com.twitter.logging.Logger
+import com.twitter.util._
+import io.buoyant.consul.v1._
+import io.buoyant.namer.{InstrumentedActivity, InstrumentedVar}
+
+class ConsulDtabCache(
```

Comment:
there seems to be a large amount of overlap here with ConsulDtabStore.  Is there a way to refactor things to share code between the two of them?  Or even better, can the Consul interpreter just use a ConsulDtabStore under the hood?

Reply:
A bonus of this approach is that it helps to ensure compatibility between the consul dtab store and the consul interpreter.

Reply:
Yea, that is correct. My dilemma was that I wanted to just use the `observe` method present in `ConsulDtabStore`, but my options were either to use the entire class or create a shared class that could be used by the consul interpreter and consul dtab storage. I didn't want to use the `ConsulDtabStore` class since it had other  CRUD operations for dtabs which I didn't want to include. I also had issues creating a shared class in that it would require me to restructure the dependencies for the consul, namer and linkerd modules. That would have caused a huge diff. I was thinking about following up with a separate PR that does the actual restructuring but probably the best bet is to just reuse `ConsulDtabStore` in the consul interpreter module.